aten/src/ATen/Quantizer.h

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <c10/core/QScheme.h>
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+#include <ATen/core/Tensor.h>
+
+#include <cmath>
+#include <memory>
+
+// TODO: move to c10 namespace after we
+// unified caffe2::Tensor and at::Tensor
+
+namespace at {
+
+using QTensor = Tensor;
+using RealTensor = Tensor;
+
+/**
+ * Quantizer is the class for storing all the information
+ * that's necessary to perform quantize and dequantize
+ * operation.
+ *
+ * We might have different types of quantization schemes and this is
+ * the base class for all quantizers.
+ *
+ * QTensorImpl will hold a pointer to Quantizer so that we can support
+ * different quantization schemes on Tensor.
+ *
+ * For example, the most common quantization scheme, Affine Quantization, requires
+ * scale and zero_point as parameters, we'll store scale and zero_point inside
+ * the instance and we can use it to quantize a float Tensor or dequantize a
+ * quantized Tensor.
+ *
+ * When you add new types of leaf Quantizer class, please also
+ * make sure to add a corresponding QScheme enum since
+ * they should have one to one mapping.
+ */
+struct C10_API Quantizer {
+  c10::QScheme qscheme_;
+  virtual ~Quantizer() {}
```

Comment:
for all header-only classes, move the destructor definition to .cpp. Otherwise compiler will make vtable a weak symbol and it can lead to ASAN's false-positives. Same applies to the classes below